Runner/JobManager.py

```diff
@@ -0,0 +1,154 @@
+        template = ""
+        with open(self.pool_template_file) as f: 
+            template = json.load(f)
+        
+        # set extra license if needed 
+        if self.application_licenses is not None:
+            template["pool"]["applicationLicenses"] = self.application_licenses.split(",")
+
+        # List of the pools
+        all_pools = [p.id for p in batch_service_client.pool.list()]
+        
+        Utils.set_template_name(template, self.pool_id)
+        if(self.pool_id not in all_pools):
+            pool_json = batch_service_client.pool.expand_template(template)
+            pool = batch_service_client.pool.poolparameter_from_json(pool_json)
+            loop = asyncio.get_event_loop()
+            await loop.run_in_executor(None, functools.partial(batch_service_client.pool.add, pool))
+        else:
+            print('pool [{}] already exists'.format(self.pool_id))
+
+    async def upload_assets(self, blob_client):
+        print("uploading asset")
+        loop = asyncio.get_event_loop()
+        input_container_name = self.job_id
+
+        # Create input container 
+        await loop.run_in_executor(None, functools.partial(blob_client.create_container, "fgrp-"+input_container_name, fail_on_exist=False))    
+        #blob_client.create_container("fgrp-"+input_container_name, fail_on_exist=False)
+        output_container_name = self.job_id+'-output'
+
+        # Create output container 
+        #blob_client.create_container("fgrp-"+output_container_name, fail_on_exist=False)
+        await loop.run_in_executor(None, functools.partial(blob_client.create_container, "fgrp-"+output_container_name, fail_on_exist=False))
+
+        scenefile = Utils.get_scene_file(self.parameters_file)
+
+        full_sas_url_input = 'https://{}.blob.core.windows.net/{}?{}'.format(blob_client.account_name, input_container_name, Utils.get_container_sas_token(blob_client, input_container_name, ContainerPermissions.READ + ContainerPermissions.LIST))
+        full_sas_url_output = 'https://{}.blob.core.windows.net/{}?{}'.format(blob_client.account_name, output_container_name, Utils.get_container_sas_token(blob_client, output_container_name, ContainerPermissions.READ + ContainerPermissions.LIST))
+        
+        # Set the storage info for the container. 
+        self.storage_info = Utils.StorageInfo(input_container_name, output_container_name, full_sas_url_input, full_sas_url_output)        
+        print("storage_info", self.storage_info)
+
+        # Upload the asset file that will be rendered and 
+        for file in os.listdir("Assets"):        
+            if scenefile == file:
+                #(blob_client, "fgrp-"+input_container_name, os.getcwd()+"\\Assets\\"+file)
+                await loop.run_in_executor(None, functools.partial(Utils.upload_file_to_container, blob_client, "fgrp-"+input_container_name, os.getcwd()+"\\Assets\\"+file))
+
+        
+
+    async def validate(self, batch_service_client):
```

Comment:
:bulb: **Important** Please note: Sample below is from async to Blocking call: using `run_in_executor`:

![image](https://user-images.githubusercontent.com/6233295/46119318-804c5280-c25e-11e8-83f2-e971fa7fabff.png)

```
def blocking_io(arg1="hulk", arg2="smash"):
    monty = "{}{}".format(arg1, arg2)
    print("----- {}".format(monty))
    return monty
async def async_foo():
    loop = asyncio.get_event_loop()
    result = await loop.run_in_executor(
        None, blocking_io, "IronMan")
    print('default thread pool', result)
    return result
def foo():
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
    result = loop.run_until_complete(async_foo())
    print("returned result is - {}".format(result))
    return result
avengers = foo()
----- IronMansmash
default thread pool IronMansmash
returned result is - IronMansmash

```

Reply:
📔  -> follow up ping on this, is this sorted, I checked recent commit seems like not catering the exact use of async used between : `main call -> to Validate -> Wait` method. Play around and please let me know and we can sit down and make small changes.

💡 : Future reference only: Best for these scenarios moving forwards will be make small test files add UTs for your scenario so that your functs are contolled.